src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/ClientConstructorAnalyzer.cs

```diff
@@ -24,18 +24,13 @@ private bool IsClientOptionsParameter(IParameterSymbol symbol)
             {
                 return false;
             }
-            var clientOptionsType = GetOptionsTypeName(symbol.ContainingType);
-            return symbol.Type.Name == clientOptionsType;
+            return symbol.Type.Name.EndsWith(ClientOptionsAnalyzer.ClientOptionsSuffix);
```

Comment:
Is it worth checking if the type derives from `ClientOptions`?

Reply:
I though about it but not deriving from ClientOptions is just not so easy mistake to make.